File: dashboard_react.client/src/types/OrderResponse.ts
import { CatalogueResponse } from "./CatalogueResponse";
import { CustomerAddressResponse } from "./CustomerAddressResponse";
import { CustomerResponse } from "./CustomerResponse";
import { ProductResponse } from "./ProductResponse";

export interface OrderResponse {
  id: string;
  customerId: string;
  customerDirectionId: string;
  orderDate: Date;
  deliveryDate: any;
  paymentTypeId: string;
  orderStateId: string;
  total: number;
  orderDetails: OrderDetailResponse[];
  createdAt: string | null;
  updatedAt?: string | null;
  createdBy: string | null;
  updatedBy?: string | null;
  customer: CustomerResponse;
  customerDirection: CustomerAddressResponse;
  paymentType: CatalogueResponse;
  orderState: CatalogueResponse;
}

export interface OrderDetailResponse {
  numberLine: number;
  productId: string;
  productName: string;
  quantity: number;
  unitPrice: number;
  totalLine: number;
  product: ProductResponse;
}

export interface OrderDetailLine {
  id: string;
  numberLine: number;
  productId: string;
  productName: string;
  quantity: number;
  unitPrice: number;
  totalLine: number;
}

export interface OrderDetailLineRequest {
  productId: string;
  productName: string;
  quantity: number;
  unitPrice: number;
}

File: Entities/Response/OrderResponse.cs

namespace Entities.Response
{
    public class OrderResponse
    {
        public string Id { get; set; } = string.Empty;
        public string CustomerId { get; set; } = string.Empty;
        public string OrderDate { get; set; } = string.Empty;
        public string DeliveryDate { get; set; } = string.Empty;
        public string PaymentTypeId { get; set; } = string.Empty;
        public string OrderStateId { get; set; } = string.Empty;
        public decimal Total { get; set; }
        public List<OrderDetailResponse> OrderDetails { get; set; } = new List<OrderDetailResponse>();
        public string CreatedAt { get; set; } = string.Empty;
        public string? UpdatedAt { get; set; }
        public string CreatedBy { get; set; } = string.Empty;
        public string? UpdatedBy { get; set; }

        public virtual CustomerResponse? Customer { get; set; }
        public virtual CatalogueResponse? PaymentType { get; set; }
        public virtual CatalogueResponse? OrderState { get; set; }
    }
}
